src/rgw/services/svc_bi_rados.cc

```diff
@@ -163,23 +163,23 @@ static void get_bucket_index_objects(const string& bucket_oid_base,
 }
 
 static void get_bucket_instance_ids(const RGWBucketInfo& bucket_info,
-                                    int shard_id,
+                                    int num_shards, int shard_id,
```

Comment:
why do you pass "num_shards" if this could be fetched from "bucket_info"?

Reply:
this function was hard-coded to use num_shards from the `current_index` layout, but this function is called by `open_bucket_index()` which takes an arbitrary `rgw::bucket_index_layout_generation` so we need to use the shards from that index layout instead of `current_index`

this is relevent now because we can request `RGWOp_BILog_Info` for any log generation; we convert that to an index layout with `log_to_index_layout()` and pass that to `RGWRados::get_bucket_stats()` which calls into `open_bucket_index()`

without this change, we hit an assert in `RGWRados::get_bucket_stats()` because `open_bucket_index()` returns a different number of `bucket_objs` (based on the given index's num_shards) vs `bucket_instance_ids` (based on current_index num_shards)

Reply:
thanks. I should have changed this as part of the bilog pr.